sdk/core/core-http/lib/atomError.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { Constants } from "./util/constants";
+import { HttpOperationResponse } from "./httpOperationResponse";
+
+/**
+ * Represents custom `Error` information returned by ATOM based services.
+ */
+export class AtomError extends Error {
+  code?: string;
+  statusCode?: number;
+  additionalProperties?: any;
+  requestId?: string;
+
+  constructor(message: string) {
+    super(message);
+    this.additionalProperties = {};
+    Object.setPrototypeOf(this, AtomError.prototype);
+  }
+}
+
+/**
+ *
+ * Utility to help construct the normalized `AtomError` object based on given `errorBody`
+ * data and other data present in the received `response` object.
+ *
+ * @param errorBody
+ * @param response
+ */
+export function buildAtomError(errorBody: any, response: HttpOperationResponse): AtomError {
+  if (typeof errorBody === "string") {
+    return new Error(errorBody);
+  }
+  const normalizedError: any = {};
+  const odataErrorFormat = !!errorBody["odata.error"];
+  const errorProperties =
+    errorBody.Error || errorBody.error || errorBody["odata.error"] || errorBody;
+
+  if (odataErrorFormat) {
+    Object.keys(errorProperties).forEach((property: string) => {
+      let value = null;
+      if (
+        property === Constants.ODATA_ERROR_MESSAGE &&
+        typeof errorProperties[Constants.ODATA_ERROR_MESSAGE] !== "string"
+      ) {
+        if (errorProperties[Constants.ODATA_ERROR_MESSAGE][Constants.ODATA_ERROR_MESSAGE_VALUE]) {
+          value =
+            errorProperties[Constants.ODATA_ERROR_MESSAGE][Constants.ODATA_ERROR_MESSAGE_VALUE];
+        } else {
+          value = "missing value in the message property of the odata error format";
+        }
+      } else {
+        value = errorProperties[property];
+      }
+      normalizedError[property.toLowerCase()] = value;
+    });
```

Comment:
The above would throw an error if `(errorProperties[Constants.ODATA_ERROR_MESSAGE])` was null or undefined

I would simplify the nesting above
```javascript
let value = errorProperties[property];
if (value && typeof value !== "string" && property === Constants.ODATA_ERROR_MESSAGE ) {
      value = value[Constants.ODATA_ERROR_MESSAGE_VALUE]
          || "missing value in the message property of the odata error format";
  }
```